qt/scientific_interfaces/ISISReflectometry/GUI/MainWindow/MainWindowPresenter.cpp

```diff
@@ -195,7 +196,15 @@ void MainWindowPresenter::notifyLoadBatchRequested(int tabIndex) {
   auto filename = QFileDialog::getOpenFileName();
   if (filename == "")
     return;
-  auto map = MantidQt::API::loadJSONFromFile(filename);
+  QMap<QString, QVariant> map;
+  try {
+    map = MantidQt::API::loadJSONFromFile(filename);
+  } catch (const std::runtime_error &e) {
+    QMessageBox::warning(
```

Comment:
I just realised this is the presenter :man_facepalming:.

Following the MVP pattern strictly your view should have a function to show the message box with a string parameter and the presenter would call this.

Reply:
Oops 🤦‍♂